### PR TITLE
feat(Level generator): bare essentials

### PR DESCRIPTION
**Pull request by DawidYerginyan**
- New Dungeon scene
- Dungeon path
- Dungeon generation barebones (will be based on cutting the path on pieces and changing lightning)
- Awesome lightning control
- Oh and also wrote editor script to update all materials in scene, just because I can. **And in reality, had a need to update unlinked materials on `1,800+` objects. :)**

### CODE REVIEW

**Review by aleksanderujek — APPROVED** [review]

File: Darkest Crusade/Assets/Dungeon/MainCamera.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    public float clampLeft;
+    public float clampRight;
+
+    private float cameraX;
+
+    void Start () {
+        cameraX = transform.position.x;
+	  }
+	
+	void Update () {
```
Comment (fyku1) [inline Darkest Crusade/Assets/Dungeon/MainCamera.cs]: This camera control script will not working on phone, please fix it

**Review by fyku1 — CHANGES_REQUESTED** [review]

File: Darkest Crusade/Assets/Dungeon/MainCamera.cs (hunk 1 above)
Comment (DawidYerginyan) [reply Darkest Crusade/Assets/Dungeon/MainCamera.cs]: 👍

File: Darkest Crusade/Assets/Dungeon/DungeonManager.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DungeonManager : MonoBehaviour {
+  public int[] dungeonBounds = { 0, 450 };
+  public int minDungeonlength = 50;
+  public int maxDungeonLength = 300;
+  public Material _material;
```
Comment (SebastianWolski) [inline Darkest Crusade/Assets/Dungeon/DungeonManager.cs]: Please, keep naming convention consistent over whole project.

File: Darkest Crusade/Assets/Dungeon/MainCamera.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MainCamera : MonoBehaviour {
+    public float speed;
+    public float clampLeft;
+    public float clampRight;
+
+    private float cameraX;
```
Comment (SebastianWolski) [inline Darkest Crusade/Assets/Dungeon/MainCamera.cs]: Please, keep naming convention consistent over whole project.

File: Darkest Crusade/Assets/Dungeon/DungeonManager.cs (hunk 2 above)
Comment (fyku1) [inline Darkest Crusade/Assets/Dungeon/DungeonManager.cs]: Please, keep naming convention consistent over whole project.

File: Darkest Crusade/Assets/Dungeon/MainCamera.cs (hunk 3 above)
Comment (fyku1) [inline Darkest Crusade/Assets/Dungeon/MainCamera.cs]: Please, keep naming convention consistent over whole project.

**Review by fyku1 — APPROVED** [review]

**Pull request merged**